libs/extensions/tabular/exec/src/lib/csv-interpreter-executor.ts

```diff
@@ -53,56 +55,51 @@ export class CSVInterpreterExecutor extends AbstractBlockExecutor<
       quote: enclosing,
       escape: enclosingEscape,
     };
-    const csvData = await parseAsCsv(file.content, parseOptions);
-
-    if (E.isLeft(csvData)) {
-      return Promise.resolve(
-        R.err({
-          message: `CSV parse failed in line ${csvData.left.lineNumber}: ${csvData.left.error.message}`,
-          diagnostic: { node: context.getCurrentNode(), property: 'name' },
-        }),
-      );
-    }
-    const sheet = new Sheet(csvData.right);
-
-    context.logger.logDebug(`Parsing raw data as CSV-sheet successful`);
-    return Promise.resolve(R.ok(sheet));
-  }
-}
-
-async function parseAsCsv(
-  lines: string[],
-  parseOptions: ParserOptionsArgs,
-): Promise<E.Either<{ error: Error; lineNumber: number }, string[][]>> {
-  let lineNumber = 1;
-  const rows: string[][] = [];
-  for await (const line of lines) {
-    const rowParseResult = await parseLineAsRow(line, parseOptions);
-    if (E.isLeft(rowParseResult)) {
-      return E.left({ error: rowParseResult.left, lineNumber });
-    }
-    rows.push(rowParseResult.right);
-
-    ++lineNumber;
+    return parseAsCSV(file.content, parseOptions).then(
+      (csvData) => {
+        context.logger.logDebug(`Parsing raw data as CSV-sheet successful`);
+        return R.ok(new Sheet(csvData));
+      },
+      (reason) => {
+        assert(typeof reason === 'string');
+        return R.err({
+          message: reason,
+          diagnostic: {
+            node: context.getCurrentNode(),
+            property: 'name',
+          },
+        });
+      },
```

Comment:
Wow I didn't know that you could add a second parameter to `.then` :D I still prefer using `.catch` because it think has more semantic value, but I don't mind using it.